src/gen/mod.rs

```diff
@@ -0,0 +1,325 @@
+                        dart_args.append(quote!(Pointer<RustCallStatus>));
+                    }
+
+                    (native_args, dart_args)
+                };
+
+                let lookup_fn = quote! {
+                    _dylib.lookupFunction<
+                        $native_return_type Function($(&native_args)),
+                        $(&dart_return_type) Function($(&dart_args))
+                    >($(format!("\"{fun_name}\"")))
+                };
+
+                definitions.append(quote! {
+                    late final $dart_return_type Function($dart_args) $fun_name = $lookup_fn;
+                });
+            }
+
+            definitions
+        }
+
+        quote! {
+            library $package_name;
+
+            $(type_helper_code) // Imports, Types and Type Helper
+
+            $(functions_definitions)
+
+            class _UniffiLib {
+                _UniffiLib._();
+
+                static final DynamicLibrary _dylib = _open();
+
+                static DynamicLibrary _open() {
+                  if (Platform.isAndroid) return DynamicLibrary.open($(format!("\"lib{libname}.so\"")));
+                  if (Platform.isIOS) return DynamicLibrary.executable();
+                  if (Platform.isLinux) return DynamicLibrary.open($(format!("\"lib{libname}.so\"")));
+                  if (Platform.isMacOS) return DynamicLibrary.open($(format!("\"lib{libname}.dylib\"")));
+                  if (Platform.isWindows) return DynamicLibrary.open($(format!("\"{libname}.dll\"")));
+                  throw UnsupportedError("Unsupported platform: ${Platform.operatingSystem}");
+                }
+
+                static final _UniffiLib instance = _UniffiLib._();
+
+                $(uniffi_function_definitions(self.ci))
+
+                static void _checkApiVersion() {
+                    final bindingsVersion = $(self.ci.uniffi_contract_version());
+                    final scaffoldingVersion = _UniffiLib.instance.$(self.ci.ffi_uniffi_contract_version().name())();
+                    if (bindingsVersion != scaffoldingVersion) {
+                      throw UniffiInternalError.panicked("UniFFI contract version mismatch: bindings version $bindingsVersion, scaffolding version $scaffoldingVersion");
+                    }
+                }
+
+                static void _checkApiChecksums() {
+                    $(for (name, expected_checksum) in self.ci.iter_checksums() =>
+                        if (_UniffiLib.instance.$(name)() != $expected_checksum) {
+                          throw UniffiInternalError.panicked("UniFFI API checksum mismatch");
+                        }
+                    )
+                }
+            }
+
+            void initialize() {
+                _UniffiLib._open();
+            }
+
+            void ensureInitialized() {
+                _UniffiLib._checkApiVersion();
+                _UniffiLib._checkApiChecksums();
+            }
```

Comment:
why two functions? Let's wrap it into one.
```suggestion
            void initialize() {
                _UniffiLib._open();
                _UniffiLib._checkApiVersion();
                _UniffiLib._checkApiChecksums();
            }
```